### ISSUE TITLE
DataTable: incorrect html rendering in case of scrollable=true and dir=rtl

### ISSUE THREAD

**Issue by bvfalcon** [primefaces/primefaces#11332]
### Describe the bug

For DataTable component in case of scrollable=true and dir=rtl css-property margin of header and footer is incorrect rendered by JavaScript.

Examples:
1) Right after initialization
![scroll-rtl-1](https://github.com/primefaces/primefaces/assets/49613576/783a490a-35ed-496e-bb27-85e4bb895c14)
Green selection -- correct: scrollbar must be on the left  of table.
Red selection -- incorrect: for header (and footer) must be set CSS-property margin-left, not margin-right.

2) After use scrollbar
![scroll-rtl-2](https://github.com/primefaces/primefaces/assets/49613576/ca4da2d8-dd7d-448d-a11f-e9ce87d0e4f7)
Red selection -- incorrect: margin-right has negative value, but must be margin-left and positive

### PrimeFaces edition

Community

### PrimeFaces version

11.0.0

### Theme

glass-x

### JSF implementation

Mojarra

### JSF version

2.2

### Java version

OpenJDK 17

### Browser(s)

Chromium

**Comment by melloware** [comment]
Have you tried 13.0.5?  A lot of RTL fixed have been made.

**Comment by melloware** [comment]
It looks like its working on the ShowcasE: https://www.primefaces.org/showcase/ui/data/datatable/scroll.xhtml?jfwid=e9b06

Then switch language to Arabic or other RTL language: 
![image](https://github.com/primefaces/primefaces/assets/4399574/93ec1775-efa5-4799-94a1-95060654c268)

**Comment by NicolaIsotta** [comment]
I think the old themeroller thems, including glass-x, are not supported from the new PrimeFaces versions. Have you tried one of the built-in themes?

**Comment by bvfalcon** [comment]
Thanks a lot, colleges, for a rapid answers!
Yes, with version 13.0.5 DataTable looks much better, but some artifacts exists. I need a little time to provide additional research. These ideas about Showcase with Arabian and old theme are very viable and I must to try them. It will be the best variant, if I could close this issue without change Primefaces core code.
I have done also some research in DataTable JavaScript library and [made a little modifications](https://github.com/bvfalcon/primefaces/commits/issue-11332/), but I hope these modifications will be superfluous.
So, I need a pair of days to research. After that I inform of my results. Once again thanks for a help!

**Comment by melloware** [comment]
ok just let us know if you have any tweaks to make.

**Comment by bvfalcon** [comment]
[Demo sample](https://github.com/bvfalcon/primefaces-datatable-scrollable-rtl-test) is prepared.
![scroll-rtl-3](https://github.com/primefaces/primefaces/assets/49613576/47847d1e-6adc-42df-9463-aa01d6e5dfa2)
Some important things. Bug exists in last version too, but:
1) It presents not in all browser equally.
This image is from Chromium, In Firefox is all right. The reason is, that different browsers show scroll different. Chromium shows scrolling right now, after page loading. But Firefox -- only then mouse is over component with scrolling. Therefore [this JS-function](https://github.com/primefaces/primefaces/blob/eb74db5d583d725700d1bd34a0095c8a98be832a/primefaces/src/main/resources/META-INF/resources/primefaces/core/core.js#L984) working different in these browsers: for Firefox scrollbarWidth is forever "0".
2) New DataTable theme design hides a bug.
In PrimeFaces 8 ([Showcase](https://www.primefaces.org/showcase-v8/ui/data/datatable/scroll.xhtml)) default DataTable presentation is table with borders. In version 13 - without borders. Absence of columns borders hides the bug. That's why I need to change the first column color of example to aquamarine: without this the problem is not visible.

So, the bug exists, but presents only a number of coincidences appear simultaneously: scrollable, RTL, browser and theme.
